### PR TITLE
Week 4- Weather app Etna 

### PR DESCRIPTION
**Pull request by Caracal23**
## Netlify link
[Netlify](https://earnest-gingersnap-d88be8.netlify.app/)

### CODE REVIEW

**Review by Martin-Joensson — COMMENTED** [review]
Great work Etna! I think you chose the design bravely and executed it very well! 

I can see a features-branch, did you do all your features in one branch or did you split them up and than deleted them? 

Your site functions well regardless of screen size, I think your might be able to tweak it for bigger screen. Right now there are very much white-space in the forecast for example. Making it harder to track the line you are on.

In terms of design I think the tweaks you did to accomidate the searchbar for example was really well done. It does not feel out of place. I also like that you used openWeathers Icons since the ones provided were a bit to few.

I'm super impressed and I just have to say: Great job once again and Go team Moon! =)

File: script.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+const API_KEY = "07b1e3ac4ed3c5e09fa788587b7c0a21";
+const DEFAULT_CITY = "Stockholm";
+
+//const URL = `${BASE_URL}?q=${DEFAULT_CITY}&units=metric&APPID=${API_KEY}`;
+//const predictURL = `${forecastURL}?q=${DEFAULT_CITY}&appid=${API_KEY}&units=metric`;
+//console.log(predictURL);
+//console.log(URL);
+
+//Current weather
+const currentWeather = (city) => {
```
Comment (HIPPIEKICK) [inline script.js:23]: Remember to name functions after what they do, e.g. fetchCurrentWeather

File: script.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        response.name,
+        response.main.temp,
+        response.weather[0].description
+      );
+      showSunriseAndSunset(response.sys.sunrise, response.sys.sunset);
+
+      descriptionContainer.innerHTML = "";
+      descriptionContainer.innerHTML += `<img src="https://openweathermap.org/img/wn/${response.weather[0].icon}@2x.png">`;
+    });
+};
```
[the remark this answers is not in the document]
Comment (HIPPIEKICK) [reply script.js:38]: Agree!

File: script.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+// Forecast days ahead
+const showForecast = (city) => {
```
Comment (HIPPIEKICK) [inline script.js:41]: Great name!

File: script.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+};
+
+// Button function
+const toggleForecast = () => {
+  if (forecastContainer.innerHTML != "") {
+    forecastContainer.innerHTML = "";
+  } else {
+    showForecast();
+  }
+};
```
Comment (HIPPIEKICK) [inline script.js:92]: Nice! Have a look at how the button gets squeezed in one of the modes though

File: script.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      showOvercast(
+        response.name,
+        response.main.temp,
+        response.weather[0].description
+      );
+      showSunriseAndSunset(response.sys.sunrise, response.sys.sunset);
+
+      descriptionContainer.innerHTML = "";
+      descriptionContainer.innerHTML += `<img src="https://openweathermap.org/img/wn/${response.weather[0].icon}@2x.png">`;
+    });
```
Comment (HIPPIEKICK) [inline script.js:37]: Some kind of error handling would be good if the user searches for a city that doesn't exist

File: script.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    showForecast(searchInputContainer.value);
+  }
+};
+
+//Event listeners
+hiddingButtonContainer.addEventListener("click", toggleForecast);
+searchFormContainer.addEventListener("submit", citySearch);
+
+//Function unixtime to weekday
+const getWeekDay = (unixtime) => {
```
Comment (HIPPIEKICK) [inline script.js:110]: Could also be dealt with using indexes

File: assets/.DS_Store
Comment (AntonellaMorittu) [inline assets/.DS_Store:1]: please delete and add to .gitignore 👍

File: .DS_Store
Comment (AntonellaMorittu) [inline .DS_Store:1]: 🧹 🧹 🧹

**Review by AntonellaMorittu — APPROVED** [review]
Hi Etna, very good job this week! Your code is well structured, the app is working well, you completed the basic requirements and added a search functionality, cool! One comment is about the forecast:

- The app should have: 4-day forecast

You are displaying all 5 days when we requested 4, not a big deal. Fix this and you're good to go:) Be careful when in mobile mode the button icon is squeezed and warped.  

Keep up the great work 🌞

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Good job this week Etna! Please have a look at how your app looks in 1600px width as per this requirement:
- Make your app responsive (it should look good on devices from 320px width up to 1600px)

I suggest you to put a max-width or in some other way add some padding to make it less stretched out:
Skärmavbild 2024-03-07 kl  15 58 43

**Review by AntonellaMorittu — APPROVED** [review]

**Pull request closed without merging**